app/views/layouts/application.html.slim

```diff
@@ -46,7 +46,7 @@ html lang="#{I18n.locale}" class='no-js'
     .site-wrap
       = render 'shared/i18n_mode' if FeatureManagement.enable_i18n_mode?
       = render 'shared/usa_banner'
-      - if user_fully_authenticated?
+      - if user_fully_authenticated? && !session[:sp].present?
```

Comment:
you are right! this logic needs to be tweaked. will return to this once we get clarity on the issues you raised in the GH issue for this PR